osu.Game/Rulesets/Mods/ModTimeRamp.cs

```diff
@@ -56,9 +62,17 @@ public virtual void ApplyToBeatmap(Beatmap<T> beatmap)
         public virtual void Update(Playfield playfield)
         {
             var absRate = Math.Abs(FinalRateAdjustment);
-            var adjustment = MathHelper.Clamp(absRate * ((clock.CurrentTime - beginRampTime) / finalRateTime), 0, absRate);
+            applyAdjustment(MathHelper.Clamp(absRate * ((clock.CurrentTime - beginRampTime) / finalRateTime), 0, absRate));
+        }
+
+        private void applyAdjustment(double adjustment)
+        {
+            var localAdjust = 1 + Math.Sign(FinalRateAdjustment) * adjustment;
 
-            pitchAdjust.PitchAdjust = 1 + Math.Sign(FinalRateAdjustment) * adjustment;
+            if (clock is IHasPitchAdjust tempo)
+                tempo.PitchAdjust = baseAdjust * localAdjust;
```

Comment:
I'm not sure why you've done it this way, seems like it's hinging on a silver thread depending on ordering of application of mods.

I would rather do the following:
```
ApplyToClock():
    lastAppliedAdjustment = 1

applyAdjustment():
    tempo.PitchAdjust /= lastAppliedAdjustment
    tempo.PitchAdjust *= localAdjust
    lastAppliedAdjustment = localAdjust
```

Reply:
In this case it's not a big issue since this mod is incompatible with the other rate adjustment mods, but it's worth a consideration because I hope that the mods would stack nicely.

Reply:
I tried the way you said at first but switched for some reason. Don't remember why, will have to revisit.